Update docs for milestone v0.5.0

tpatki opened the pull request:
See issue https://github.com/LLNL/variorum/issues/238 . We will complete some relevant tasks form this issue for this upcoming milestone. 

Also fixes #220.

Docs viewable here: https://variorum.readthedocs.io/en/2022_update_docs/

tpatki committed:
Update docs to separate Intel  requirements from general requirements from Quick Start page

tpatki committed:
Add Intel and IBM requirement sections, update BuildingVariorum and QuickStart accordingly

tpatki committed:
Minor updates to IBM requirements

tpatki committed:
Add a requirements section for each vendor

tpatki committed:
Resrtucture page tree

tpatki committed:
Add Powmon documentation, edit authors, add Ice lake to Readme

tpatki committed:
Minor fix for formatting

tpatki committed:
Remove redundancies from BuildingVariorum.rst, restructure the toctree

tpatki committed:
Minor formatting edit

tpatki committed:
Remove duplicated text

tpatki committed:
Reorder again

tpatki committed:
formatting fix

tpatki committed:
Make main page more compact, remove space. Chaneg contributors to key developers as contibutor list is long and many are not active

tpatki committed:
Formatting

tpatki committed:
Reorder page tree, add empty powerstack page

tpatki committed:
Add ECP Acknowledgment. Use full acknowledgment as it mentions software

tpatki committed:
Testing my PowerStack.rst is not getting picked up

tpatki committed:
Testing my PowerStack.rst is not getting picked up

tpatki committed:
Initial Contributors List

tpatki committed:
Add missing contributors file

tpatki committed:
Reorder page tree, create Argo page link

tpatki committed:
Add empty powerapi page

tpatki committed:
More page restructuring

tpatki committed:
More page restructuring

tpatki committed:
More page restructuring

tpatki committed:
More page restructuring

tpatki committed:
More page restructuring

tpatki committed:
WIP Jun 9

tpatki committed:
Update README to point to new contributors list, add initial text for Argo page

tpatki committed:
Initial PowerAPI text

tpatki committed:
Minor fix

tpatki committed:
Minor fix

tpatki committed:
Minor fix

tpatki committed:
Minor fix

tpatki committed:
Initial PowerStack text

tpatki committed:
Fix variorum logo in docs

tpatki committed:
Minor formatting error

tpatki committed:
Image formatting

tpatki committed:
Image formatting

tpatki committed:
Image formatting

tpatki committed:
More updates to docs, add eehpc content

tpatki committed:
Change Overview to underplay libmsr, so Variorum doesn't sound like its extension, which it is not.

tpatki committed:
Update Integration doc

tpatki committed:
Run spell check

tpatki committed:
Add content to Integration document, add links to various frameworks

tpatki committed:
Fix links

tpatki committed:
Fix link for PowerStack, will only be active after merging

tpatki committed:
Fix link for PowerStack, will only be active after merging

tpatki committed:
Fix link for PowerStack, will only be active after merging

tpatki committed:
Update Contributors link

Link will be active after merge

slabasan committed:
updates to QuickStart and BuildingVariorum

- fix default values for variorum variables from top-level CMakeLists.txt
- update build instructions to create separate build and install directory

rountree commented:
* https://variorum.readthedocs.io/en/2022_update_docs/
    - Text in the tab reads "Variorum 0.4.1 documentation"
    - "machine-specific registers" -> "model-specific registers"
    - "HPC application developers" -> "application developers"
    - "for the devices contained within the node" delete?
    - "through the following" -> "by" (or "by providing")
    - "It includes" -> delete x3?
    - Add link to code of conduct?

rountree commented:
Just so it's not lost in the bulk of the previous comment, excellent work (that built on some excellent work)!

Thanks for putting in the effort here.

slabasan reviewed (approved):
LGTM, thank you for all your efforts on this!

tpatki commented:
Thank you @slabasan and @rountree for the detailed reviews. I will make these changes tomorrow and we can merge the docs then.

tpatki commented:
Capturing @Jessicat-H 's comments here so I don't forget to incorporate them

- docs Build Variorum’s Documentation OFF (~docs)
- log Enable Variorum’s logging infrastructure OFF (~docs)
- Minor thing, but I think this is a typo in the new documentation - in the table on the Building Variorum page, I assume that Default column for the log variant should say OFF (~log)?
- Also, the spack install variorum command is on that page twice which seems a little redundant







[8:17](https://llnl-performance.slack.com/archives/C03J5HFANHJ/p1655133432771199)
Also, the spack install variorum command is on that page twice which seems a little redundant
:+1:
1

slabasan committed:
fix markdown formatting for inline

tpatki committed:
6/13/22: Adressing Barry's comments. Progress before lunch.

tpatki committed:
6/13/22: Barry's comments, WIP

tpatki committed:
6/13/22: Barry's comments, minor update to Nvidia page.

tpatki committed:
Minor formatting.

tpatki committed:
Minor formatting fixes

slabasan committed:
minor updates to BuildingVariorum

- fix ~log for disabling log with spack build
- remove spack building tests command

slabasan commented on src/docs/sphinx/conf.py.in:
```diff
@@ -52,9 +52,9 @@ copyright = u'2019-2021, LLNS'
 #
 # The short X.Y version.
-version = '0.4.1'
+version = '0.5.0'
```
We have traditionally updated the version in a commit where we are bumping the release version throughout the entire project. I have already captured an update to the version here: https://github.com/LLNL/variorum/pull/251

slabasan commented on .github/workflows/github-actions.yml:
```diff
@@ -405,6 +405,14 @@ jobs:
+        if: ${{ matrix.python-version == 3.7 }}
+        run: |
+          cd ${GITHUB_WORKSPACE}
+          ./scripts/rst-format.sh
```
The rst-format.sh script is missing from this branch. Also, is the plan to format all rst files with this script before merging in this branch?

tpatki replied:
Yes, I think we should format it. I didn't add the formatting script, I thought you added it -- how do I grab the `rst-format.sh`? It doesn't seem to be committed here.

slabasan replied:
Oh did I push that commit? Let me check, I don't recall writing it, but I know i was looking into it on the drive up.

slabasan commented on README:
```diff
@@ -203,3 +202,10 @@ Variorum is released under the MIT license. For more details, see the
+a joint project of the U.S. Department of Energy's Office of Science and 
+National Nuclear Security Administration, responsible for delivering a capable 
+exascale ecosystem, including software, applications, and hardware technology, 
+to support the nation's exascale computing imperative.
```
Please also update the date updated in this file.

tpatki committed:
Revert version number change in conf.py.in

tpatki committed:
Revert version number change in conf.py

tpatki committed:
6/15/22. Address more of Barry's and Jessica's comments. Add code of conduct

tpatki committed:
Update system diagram.

tpatki committed:
6/15/22: Addressing Barry's comments. Restructing and formatting for BuildingVariorum.rst

tpatki committed:
Spack variants table not being rendered...debugging

tpatki committed:
Spack variants table not being rendered...debugging

tpatki committed:
Minor changes to QuickStart

tpatki committed:
Move descriptions of dependencies from Overview to BuildingVariorum, makes more sense there.

tpatki committed:
Make figure smaller, add apt-get to install packages

tpatki commented:
@rountree @slabasan 
I've addressed most of the comments, and where appropriate, updated the docs issue #261 to capture what didn't go in this time around. 

Two final things I need help with:
- the rstfmt.sh script needs to be committed, and run before merging this in
- We need to capture any specific versions in our dependency on hwloc, jansson and rankstr, as pointed out by @rountree.

tpatki committed:
minor update to QuickStart

tpatki committed:
Update index.rst to make it less verbose

tpatki committed:
Update VariorumTools to include links and added explanation

tpatki commented:
@rountree @slabasan
I've addressed most of the comments, and where appropriate, updated the docs issue https://github.com/LLNL/variorum/issues/261 to capture what didn't go in this time around.

Two final things I need help with:

- the rst-format.sh script needs to be committed, and run on these files to format them before merging this in
- We need to capture any specific versions in our dependency on hwloc, jansson and rankstr, as pointed out by @rountree.

tpatki committed:
Minor update to VariorumTools.rst

tpatki committed:
Update index.rst to include Argo and PowerStack

tpatki committed:
Update index.rst

tpatki committed:
Update index.rst

tpatki committed:
Figure can be bigger in Overview.rst

tpatki committed:
Change order of build options

For readability, add paths first, architectures next, mpi/openmp next and rest later.

slabasan committed:
undo rstfmt

slabasan committed:
put build spack section last

slabasan commented on src/docs/sphinx/BuildingVariorum.rst:
```diff
@@ -7,162 +7,150 @@
 
 .. code:: bash
 
+   sudo apt-get install libhwloc15 libhwloc-dev libjansson4 libjansson-dev
```
Should we be including this in the build steps? It seems specific to a given OS.

tpatki replied:
I think @rountree suggested this. I personally think it is a good idea to have it in there because it gets most people up and running quickly, and people can figure out the equivalent yum install packages.

slabasan replied:
I added a comment for this line to specify installing of variorum dependencies.

slabasan committed:
testing breathe integration for docs

slabasan committed:
remove unused rst file

slabasan committed:
start adding supported architectures to functions in variorum.h

completed: IBM, AMD, ARM, NVIDIA
in progress: Intel

slabasan commented on src/docs/sphinx/BuildingVariorum.rst:
```diff
@@ -7,162 +7,150 @@
 ###################
 
-This page provides details on how to build Variorum from source.
+Variorum can be built with ``spack`` or with CMake from source. Building Variorum creates the
```
switch spack and CMake in first sentence, since sections are flipped below

slabasan commented on src/docs/sphinx/VariorumDevel.rst:
```diff
@@ -60,13 +60,13 @@ Example
 *******
 
 As an example, to support additional Intel microarchitectures:
```
change example to point at nvidia instead of intel

slabasan committed:
add breathe dependency to CI

slabasan committed:
add API pages to index.rst

slabasan committed:
use custom doxygen header for supported arch

slabasan committed:
finish supported functions for Intel

slabasan committed:
small updates

- switch cmake and spack in Building Varioum section to match the order in
  which they are detailed
- add comment about installing jansson and hwloc dependencies to bash commands
  showing build/install of variorum
- remove trailing whitespace in BuildingVariorum and VariorumDevel
- use nvidia as an example for extending to additional microarchitectures,
  simpler to digest the code structure

slabasan committed:
fix italics to inline code annotation

slabasan committed:
test doxygen unsupported headline

for each API, we can document which architectures cannot be supported due to
feature limitations

slabasan committed:
update title of api toggle functions

slabasan committed:
fix docs build warnings

- fix path to logo
- change name of module recordings to be unique

slabasan committed:
update variorum.h doxygen comments

slabasan committed:
add links to API pages in QuickStart and VariorumAPI pages

slabasan committed:
fix referenced variorum header

slabasan committed:
spellcheck

slabasan committed:
change P9 to Power9 in header doxygen comments

tpatki committed:
Update order of print and cap function docs

tpatki committed:
Update order of print and cap function docs.

tpatki committed:
Order of APIs in QuickStart

slabasan committed:
rearrange print functions

slabasan committed:
no IBM P9 support for cap_each_socket_power_limit

slabasan committed:
update doxyfile

slabasan committed:
add comment about print vs print_verbose in print api page

slabasan committed:
fix boldface
